zephyr/include/rtos/alloc.h

```diff
@@ -146,4 +146,26 @@ static inline void heap_trace_all(int force) {}
 
 /** @}*/
 
+#if CONFIG_USERSPACE
+/**
+ * Returns the start address of shared memory heap.
+ *
+ * @return pointer to shared memory heap start
+ *
+ * @note Needed to add shared heap to private module memory domain.
+ */
+uintptr_t get_shd_heap_start(void);
+
+/**
+ * Returns the size of shared memory heap.
+ * @param ptr None
+ *
+ * @return pointer to shared memory heap start
+ *
+ * @note Needed to add shared heap to private module memory domain
+ */
+size_t get_shd_heap_size(void);
```

Comment:
So we will have USER and KERNEL heaps. The USER heap will contain smaller per USER memory domain heaps. e.g. if we had 2 3rd party DP modules, they would each have a USER heap with its own memory domain. 
I'm assuming this API is for the large USER heap and not the per memory domain sub USER heaps ?

If so, I would also rename to have a ```heap_user_``` API prefix.

Reply:
This memory is intended for audio buffers and will be shared between all user modules and kernel.